section11/numberformat_proposed.html

```diff
@@ -143,11 +162,19 @@ <h1>PartitionNumberPattern ( _numberFormat_, _x_ )</h1>
       </p>
 
       <emu-alg>
-        1. If _x_ is not *NaN* and _x_ < 0, then
-          1. Let _x_ be -_x_.
-          1. Let _pattern_ be _numberFormat_.[[NegativePattern]].
+        1. Let _exponent_ be 0.
+        1. If _x_ is *NaN*, then
+          1. Let _n_ be an ILD String value indicating the *NaN* value.
+        1. Else if _x_ is not a finite Number,
+          1. Let _n_ be an ILD String value indicating infinity.
         1. Else,
-          1. Let _pattern_ be _numberFormat_.[[PositivePattern]].
+          1. If _numberFormat_.[[Style]] is `"percent"`, let _x_ be 100 × _x_.
+          1. Let _exponent_ be ComputeExponent(_numberFormat_, _x_).
+          1. Let _x_ be _x_ × 10<sup>-_exponent_</sup>.
+          1. Let _formatNumberResult_ be FormatNumberToString(_numberFormat_, _x_).
+          1. Let _n_ be _formatNumberResult_.[[FormattedString]].
+          1. Let _x_ be _formatNumberResult_.[[RoundedNumber]].
+        1. Let _pattern_ be GetNumberFormatPattern(_numberFormat_, _x_, _exponent_)
```

Comment:
It's a bit surprising that you do this calculation of which pattern to use on each format. It seems like the existing formatters choose the pattern up-front, mostly. Do you think there's anything of this that would make sense to (spec-wise) "pre-compute"? This might be useful for implementations to follow a similar split.